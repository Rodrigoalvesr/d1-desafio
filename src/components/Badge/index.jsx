import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import classnames from 'classnames';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    width: 22,
    height: 22,
    background: '#E4E6F1',
    border: '1px solid #EBEEF6',
    borderRadius: '50%',
    '&.selected': {
      color: '#FFFFFF',
      background: theme.palette.primary.main,
    },
  },
  text: {
    color: '#9196AB',
    fontSize: 10,
    fontWeight: 'bold',
    '&.selected': {
      color: '#FFFFFF',
    },
  },
}));

function Badge({ selected, value, className }) {
  const classes = useStyles();

  return (
    <div className={classnames(classes.root, className, { selected })}>
      <Typography className={classnames(classes.text, { selected })}>
        {value}
      </Typography>
    </div>
  );
}

Badge.defaultProps = {
  selected: false,
};

Badge.propTypes = {
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  selected: PropTypes.bool,
  className: PropTypes.string,
};

export default Badge;
