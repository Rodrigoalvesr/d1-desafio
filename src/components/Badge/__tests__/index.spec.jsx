import React from 'react';
import renderer from 'react-test-renderer';
import Badge from '../index';

describe('Badge', () => {
  it('Renders correctly', () => {
    const component = renderer.create(<Badge value={10} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
