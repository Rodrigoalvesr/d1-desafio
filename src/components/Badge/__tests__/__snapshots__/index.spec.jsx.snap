// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <p
    className="MuiTypography-root makeStyles-text-2 MuiTypography-body1"
  >
    10
  </p>
</div>
`;
